### PR TITLE
docs: reduce costs in the background services rfc

### PR DESCRIPTION
**Pull request by Ubisoft-potato**
This PR will add an RFC (Request for Comments) that demonstrates how to implement serverless architecture for background services.

Reference: https://github.com/bucketeer-io/bucketeer/issues/612

### CODE REVIEW

File: docs/rfcs/serverless-for-background-services.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	puller := client.createPuller(ctx)
+	cctx, cancel := context.WithCancel(ctx)
+	p.cancel = cancel
+	puller.Run(cctx, func() {
+		// process events
+	})
+}
+
+func (p Persister) unsubscribe() {
+	p.cancel()
```
Comment (cre8ivejp) [inline docs/rfcs/serverless-for-background-services.md:81]: Just confirming.
Have you considered using the `unsubscribe` interface from PubSub?

Comment (Ubisoft-potato) [reply docs/rfcs/serverless-for-background-services.md:81]: Yes, but I didn't find `unsubscribe` interface in PubSub Goalang api.

Comment (Ubisoft-potato) [reply docs/rfcs/serverless-for-background-services.md:81]: According to the PubSub doc [Receive](https://pkg.go.dev/cloud.google.com/go/pubsub#Subscription.Receive) function : 
```
//The standard way to terminate a Receive is to cancel its context:
cctx, cancel := context.WithCancel(ctx)
err := sub.Receive(cctx, callback)
```

**Review by masaaania — APPROVED** [review]
LGTM!

**Pull request merged**